### PR TITLE
Fix evaluation of two tests

### PR DESCRIPTION
**Pull request by symphorien**
###### Motivation for this change
Fix evaluation of two tests.
yabar builds fine now, but not hocker fetchdocker.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
cc @Ma27

File: nixos/tests/yabar.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import ./make-test.nix ({ pkgs, lib }:
+import ./make-test.nix ({ pkgs, lib, system }:
```
Comment (xeji) [inline nixos/tests/yabar.nix]: could you add  `system ? builtins.currentSystem` here? Many tests do this, it makes them easier to run standalone (without `release.nix`).

**Comment by xeji** [comment]
@GrahamcOfBorg test yabar

**Comment by symphorien** [comment]
Apparently waitForUnit inferferes with systemd's restarting the service. When lucky, the unit is restarted exactly when the perl script checks that the unit is alive: here is a successful run of the test:
```
yabar# [   19.885106] systemd[801]: Startup finished in 118ms.
yabar# [   20.461154] systemd[801]: Started yabar service.
yabar# [   20.462729] systemd[801]: Reached target Current graphical user session.
yabar# [   20.747167] yabar-start[857]: Please add at least one bar in the `bar-list` entry.
yabar# [   20.748808] yabar-start[857]: Exiting...
yabar: running command: journalctl -b SYSLOG_IDENTIFIER=systemd | grep 'Reached target Current graphical'
yabar: exit status 0
yabar: running command: [ -e /tmp/.X11-unix/X0 ]
yabar: exit status 0
yabar: waiting for unit ‘yabar.service’
yabar: running command: su -l bob -c $'XDG_RUNTIME_DIR=/run/user/`id -u` systemctl --user --no-pager show "yabar.service"'
yabar# [   20.935626] su[865]: Successful su for bob by root
yabar# [   20.939550] su[865]: pam_unix(su:session): session opened for user bob by (uid=0)
yabar# [   21.115810] systemd[801]: yabar.service: Service hold-off time over, scheduling restart.
yabar# [   21.119296] systemd[801]: yabar.service: Scheduled restart job, restart counter is at 1.
yabar# [   21.127598] systemd[801]: Stopped yabar service.
yabar# [   21.133216] systemd[801]: Started yabar service.
yabar# [   21.190359] su[865]: pam_unix(su:session): session closed for user bob
yabar: exit status 0
collecting coverage data
yabar: running command: test -e /sys/kernel/debug/gcov
yabar: exit status 1
syncing
yabar: running command: sync
yabar# [   21.210457] yabar-start[879]: Please add at least one bar in the `bar-list` entry.
yabar# [   21.212935] yabar-start[879]: Exiting...
yabar: exit status 0
test script finished in 29.10s
cleaning up
killing yabar (pid 600)
vde_switch: EOF on stdin, cleaning up and exiting
vde_switch: Could not remove ctl dir '/build/vde1.ctl': Directory not empty
```

Anyway, my goal is just that they evaluate.

**Comment by xeji** [comment]
> Anyway, my goal is just that they evaluate.

Fair enough, so let's merge this now and fix test failures separately.

**Comment by Ma27** [comment]
Thanks for fixing this! I'm sorry for the extra work, I just tested it with `nix-build ./nixos/tests/$test.nix`, but not from `release.nix` :/

**Comment by xeji** [comment]
@Ma27 could you have a look at the non-deterministic failure of the `yabar` test when you find the time? Not urgent but should be fixed eventually...

**Pull request merged**